File: ml-service/ml_service/reasons.py
from __future__ import annotations


def reason_code_for_candidate(
    *,
    similarity: float,
    has_deadline_soon: bool,
    overlap_count: int,
    target_year_match: bool,
) -> str:
    if has_deadline_soon:
        return "deadline_urgent"
    if overlap_count > 0 or target_year_match:
        return "interest_match"
    if similarity > 0.65:
        return "semantic_match"
    return "hybrid_match"


def reason_text(reason_code: str) -> str:
    if reason_code == "deadline_urgent":
        return "This club has an active deadline soon."
    if reason_code == "interest_match":
        return "This club overlaps with your stated interests or year fit."
    if reason_code == "semantic_match":
        return "This club is close to your recent behavior in the recommendation space."
    return "This club blends profile and behavioral signals well."

File: src/lib/recommendations/reasons.ts
export function recommendationReasonText(reasonCode: string, clubName: string) {
  switch (reasonCode) {
    case "deadline_urgent":
      return `${clubName} has an active deadline coming up soon.`;
    case "verified_popular":
      return `${clubName} is a verified club with strong recent student interest.`;
    case "popular_on_rush":
      return `${clubName} is trending with students on Rush right now.`;
    case "semantic_match":
      return `${clubName} matches your interests and past club behavior.`;
    case "interest_match":
      return `${clubName} lines up with the interests listed in your profile.`;
    case "hybrid_match":
      return `${clubName} blends your profile and recent activity signals well.`;
    default:
      return `${clubName} is a strong fit based on your recent Rush activity.`;
  }
}
